akka-actor-typed/src/main/scala/akka/actor/typed/javadsl/BehaviorInterceptor.scala

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright (C) 2018 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.actor.typed.javadsl
+import akka.actor.typed
+import akka.actor.typed.{ Behavior, Signal }
+
+abstract class BehaviorInterceptor[O, I] extends akka.actor.typed.BehaviorInterceptor[O, I] {
```

Comment:
The motivation for this is that when implementing a Java `PersistentBehavior`, you'll typically need to import `akka.actor.typed.javadsl.ActorContext`. If you are also extending `akka.actor.typed.BehaviorInterceptor` in the same source file, you'll need to declare `akka.actor.typed.ActorContext` parameters. In Java, this name conflict can only be resolved by using fully-qualified names, which is awkward and verbose.

This class is a convenience allowing you to use `akka.actor.typed.javadsl.ActorContext` to implement a `BehaviorInterceptor`.

It needs API docs, but before doing that I wanted to make sure the Akka team thinks it's a good idea to add it.

Reply:
That is a good point regarding the name conflict when implementing it inline (in same file).
Then we should add same in scaladsl for consistency, and perhaps we don't need the common type if we can provide asJava/asScala converters.

Reply:
I think it is worth having separate APIs but I also think that actually implementing an interceptor will be uncommon and a very super-user thing to do, so I don't think it is somewhat ok if it isn't the most ergonomic API.

Reply:
Given that this also came up with `ExtensibleBehavior` above, now I'm thinking maybe a better approach would be to rename the base `akka.actor.typed.ActorContext` trait to something that doesn't conflict… maybe `BaseActorContext`?

Reply:
I just had the same though :-)
We can take that forward, @TimMoore

Reply:
created separate issue for this https://github.com/akka/akka/issues/25734

Reply:
Great, thanks! I'll revert this change here.